os/include/tinyara/mm/mm.h

```diff
@@ -411,15 +411,15 @@ extern struct mm_heap_s g_kmmheap[CONFIG_KMM_NHEAPS];
 #define BASE_HEAP ((struct mm_heap_s *)((struct tcb_s*)sched_self())->uheap)
 
 #elif defined(CONFIG_BUILD_PROTECTED) && !defined(__KERNEL__)
-extern uint32_t _stext;
+extern uint32_t _sdata;
 
 #ifdef CONFIG_SUPPORT_COMMON_BINARY
 extern struct mm_heap_s *g_app_heap_table[CONFIG_NUM_APPS + 1];
 extern uint32_t g_cur_app;
 #define BASE_HEAP (g_app_heap_table[g_cur_app])
 
 #else
-#define BASE_HEAP ((struct mm_heap_s *)_stext)
+#define BASE_HEAP ((struct mm_heap_s *)_sdata)
```

Comment:
If this is right, how does imxrt work?

Reply:
1. Yes in nxp its working fine
2. It is to store the heap pointer not the direct start of heap